### ISSUE TITLE
private tables cleaning

### ISSUE THREAD

**Issue by christophemacabiau** [graphile/worker#412]
### Summary
Is there a way to clean the private tables ?
We need to create queues during the lifetime of our application. When a new queue is created, an older queue becomes useless. Is there a way to remove it from the _private_tasks table?

### Additional context
We need to target some consumers for some jobs, and the forbidden flags mechanism  is not very practical (we would need the allowed flags version). The best way we have found is to create queues listened only by the target consumers (which are temporary).

**Comment by christophemacabiau** [comment]
Do you want a PR?

**Comment by benjie** [comment]
Yes, that sounds reasonable. For now let's make it a library only method (i.e. don't put it in a migration, no database function) as this gives us more flexibility to iterate it. If you want, you could also hook it up to the CLI via e.g. `graphile-worker --cleanup` or similar. Mark it as `/** @experimental */`

Something like:

```ts
// Please don't use enums here... I hate enums in TypeScript.
type CleanupTask =
  | "GC_TASK_IDENTIFIERS"
  | "GC_JOB_QUEUES"
  | "DELETE_PERMAFAILED_JOBS";

interface WorkerUtils {
  /** @experimental */
  cleanupDb(options: {
    tasks?: CleanupTask[], // default: ["GC_TASK_IDENTIFIERS", "GC_JOB_QUEUES"]
  } = {}): Promise<void>
}
```

DELETE_PERMAFAILED_JOBS:

```sql
delete from graphile_worker._private_jobs jobs
where attempts = max_attempts
and locked_at is null;
```

GC_TASK_IDENTIFIERS:

```sql
delete from graphile_worker._private_tasks tasks
where not exists (
  select 1
  from graphile_worker._private_jobs jobs
  where jobs.task_id = tasks.id
);
```

GC_JOB_QUEUES:

```sql
delete from graphile_worker._private_job_queues job_queues
where locked_at is null and not exists (
  select 1
  from graphile_worker._private_jobs jobs
  where jobs.job_queue_id = job_queues.id
);
```

If enabled, be sure to run `DELETE_PERMAFAILED_JOBS` prior to the others since they may depend on its result.

**Comment by christophemacabiau** [comment]
Thanks for the information. I will look at this on Monday..

**Comment by benjie** [comment]
Implemented in #415

**Comment by benjie** [comment]
Released in v0.16.2

If you felt like extending this functionality further, really we shouldn't try and delete task identifiers that we know we're using - so exclude those that the current worker should understand.
